lib/galaxy/managers/users.py

```diff
@@ -278,6 +279,22 @@ def error_if_anonymous(self, user, msg="Log-in required", **kwargs):
             raise exceptions.AuthenticationFailed(msg, **kwargs)
         return user
 
+    def get_user_by_identity(self, identity):
+        """Get user by username or email."""
+        user = None
+        if VALID_PUBLICNAME_RE.match(identity):
```

Comment:
The unit test fails now, because `VALID_PUBLICNAME_RE ` requires lowercase public names. Has that always been a requirement ? If yes `assert self.user_manager.get_user_by_identity(uppercase_user.username) == uppercase_user` can be dropped from the test, otherwise this we could do:
```suggestion
        if VALID_PUBLICNAME_RE.match(identity.lower()):
```

Reply:
> The unit test fails now, because VALID_PUBLICNAME_RE requires lowercase public names. Has that always been a requirement ?

Yes, I think so. I've made an alternative suggestion below.